### ISSUE TITLE
Add pty Support for core.remote action.

### ISSUE THREAD

**Issue by FileMagic** [StackStorm/st2#5826]
I want to be able to request a pty for the core.remote action. I can not currently get a pty through the core.remote though ssh config changes due to paramiko not honoring [OpenSSH config files](https://docs.paramiko.org/en/stable/api/config.html).

I also tried using the `ssh_runner`'s `use_ssh_config ` and `ssh_config_file_path` with `RequestTTY force`. This lead me to realize that the OpenSSH config was not honored through the remote shell runner either.

So I propose that we could add something to the args of [run of paramiko_ssh](https://github.com/StackStorm/st2/blob/master/st2common/st2common/runners/paramiko_ssh.py#L379) to be passed to [Line 406](https://github.com/StackStorm/st2/blob/master/st2common/st2common/runners/paramiko_ssh.py#L406) so that we can then call this method with a `force_pty` argument.

This is my first time looking to contribute to this project so I apologize if I am missing something quite obvious here.

**Comment by armab** [comment]
Thanks for opening an issue!

Any use case examples of why `pty` would be beneficial for the `core.remote`?

**Comment by FileMagic** [comment]
My primary usage is using a CLI command that supports things being piped to them. For instance, a lot of linux bash scripts use `[ -t 0 ]` to check if the program is being ran with something being piped to it. I am willing to work on a PR for this as well.

**Comment by armab** [comment]
@StackStorm/contributors @StackStorm/maintainers Thoughts on this? ^^

**Comment by nzlosh** [comment]
Can the exact use case be explained?  I'm having trouble understanding how shell pipes would be used with the `core.remote` action.

**Comment by FileMagic** [comment]
Yep I apologize, for the delay, however, the initial problem I was having was I wanted to use a pty.



I basically want to simulate using a tty as a user while using applications that are usually cli base, but then are being used by stackstorm for automation reasons.

For an example, let's say we have an application that takes input from standard in via the cli. Many times when these applications check for input via standard in they use something similar to this stack overflow [here](https://stackoverflow.com/a/30520299) which states something similar to:

```bash
[[ -t 0 ]] && echo 'STDIN is attached to TTY'
```

Which would print `STDIN is attached to TTY` if something was piped to the statement. This seems to be pretty common practice while checking if someone is using `stdin`. So that is why I was running into this issue.
